tracks/rust/exercises/luhn/mentoring.md

```diff
@@ -0,0 +1,28 @@
+### Concepts
+
+- higher-order functions
+- iterators
+- str to digits
+
+### Reasonable solutions
+
+A reasonable solution should do the following:
+
+- use iterators rather than loops.
+- utilize higher order functions on iterators and on option types
+
+### Examples
+
+```rust
+pub fn is_valid(code: &str) -> bool {
+    code.chars()
+        .rev()
+        .filter(|c| !c.is_whitespace())
+        .try_fold((0, 0), |(sum, c), d| {
```

Comment:
I have to be honest, I haven't solved this one in rust yet, but shouldn't we use `enumerate()` instead of manually bookkeeping the current index?